ranger/config/commands.py

```diff
@@ -1959,3 +1959,46 @@ def tab(self, tabnum):
             in sorted(self.modes.keys())
             if mode
         )
+
+
+class confirm(Command):
+    """:confirm <command_to_be_confirmed> + <potential command arguments>
+
+        Can be used with any command. Ranger will ask if it should execute the
+        given command.
+    """
+
+    def execute(self):
+        from functools import partial
+
+        command = ''
+        if len(self.args) > 1 and self.arg(1) != 'chain':
+            for index in range(1, len(self.args)):
+                if index != len(self.args):
+                    command += self.args[index] + ' '
+            self.fm.ui.console.ask(
+                "Confirm execution of: %s (y/N)" % ''.join(command),
+                partial(self._question_callback, command),
+                ('n', 'N', 'y', 'Y'),
+            )
+        else:
+            arguments = self.rest(2).split(';')
+            for index, item in enumerate(arguments):
+                arguments[index] = item.strip()
+            self._chaining(len(arguments) - 1, arguments)
+
+    def _question_callback(self, command, answer):
+        if answer == 'y' or answer == 'Y':
+            self.fm.execute_console(command)
+
+    def _chaining(self, recursion_count, commands):
+        from functools import partial
+        if recursion_count >= 0:
+            command = commands[recursion_count]
+            self.fm.ui.console.ask(
```

Comment:
Why is this duplicated here?

Reply:
You mean the module import? Or do you mean I could use the commands[recursion_count] directly in the message?

Well without the module import the partial function is not part of the namespace in the chaining function and would not be executed.

Reply:
I mean the entire ask part.

Reply:
Well if I do it recursively I would have to call on that function again and again I guess?

Reply:
What does recursion have to do with duplication of code?

Reply:
Nothing directly, but I need to call the ask part in the non-recursive part and in the recursive part. Well I could pack that into another function which gets called where needed.

Reply:
Okay packed it into a helper function and thus avoiding duplicated code and the duplicated module import.

Reply:
Can't you just stow it in the non-recursive part? And then "recurse" (to a depth of 1) even for the simple case?

Reply:
A helper might be fine too. Go with whatever seems clearest.